File: run_command.scala
#!/usr/bin/env scala

import sys.process.{ ProcessLogger, Process }
import java.io.File

object Command {
  def runSeq(in: Seq[String]): (List[String], List[String], Int) = {
    val qb = Process(in)
    var out = List[String]()
    var err = List[String]()

    val exit = qb ! ProcessLogger((s) ⇒ out ::= s, (s) ⇒ err ::= s)

    (out.reverse, err.reverse, exit)
  }
  def main(args: Array[String]) {

      val cmd = Seq("ps", "waux")

      val (out, err, exit) = Command.runSeq(cmd)
      println("cmd -> " + cmd)
      println("out -> ")
      out.foreach(println(_))
      println("err -> ")
      err.foreach(println(_))
      println("exit -> " + exit)
  }
}

Command.main(args)

File: run_command.py
#!/usr/bin/env python

import subprocess

def run_bash(cmd):
	process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
	return (process.communicate()[0], process.stderr, process.returncode)


def main():
	bashCommand = "ps waux"
	(out, err, exit) = run_bash(bashCommand) 
	print "out -> "
	print out
	print "err -> "
	print err
	print "exit -> "
	print exit

if __name__=='__main__':
	main()
